### PR TITLE
Internal improvement: refactor similar getBlock code

### PR DESCRIPTION
**Pull request by CruzMolina**
Inspired by Code Climate, this lil PR aims to improve `truffle-blockchain-utils`' maintainability and reduce tech debt.

https://codeclimate.com/github/trufflesuite/truffle/compare/ccBUtils

### CODE REVIEW

File: packages/truffle-blockchain-utils/index.js [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
+  getBlockBy(NumberOrHash, value, provider, callback) {
+    defaultJSONQuery.params = [value, true];
+    if (NumberOrHash === "Number")
+      defaultJSONQuery.method = "eth_getBlockByNumber";
+    else if (NumberOrHash === "Hash")
+      defaultJSONQuery.method = "eth_getBlockByHash";
+    else
+      throw new Error(
+        `.getBlockBy() only currently supports querying a "Number" or "Hash"!`
+      );
```
Comment (adrianmcli) [inline packages/truffle-blockchain-utils/index.js]: I'd push back on having string-denoted things that alter a function's behaviour. A bit of a code smell IMO. I actually think the duplication here is not that big a deal and it's actually more complex and less readable than before.

**Comment by adrianmcli** [comment]
I'd push back on having string-denoted things that alter a function's behaviour. A bit of a code smell IMO. I actually think the duplication here is not that big a deal and it's actually more complex and less readable than before.

**Comment by CruzMolina** [comment]
Thanks for sharing your thoughts @adrianmcli! What do you think about the [first commit](https://github.com/trufflesuite/truffle/pull/2100/commits/8733cfb56a6b2e4ba8fadea9050e0211e1118a2c)?

**Comment by adrianmcli** [comment]
@CruzMolina I still think that would be a type of in-direction. I would prefer duplication in this case because it's clearer to an unfamiliar reader of the code what is happening.

**Comment by CruzMolina** [comment]
Turns out we can simplify w/o duplication https://github.com/CruzMolina/truffle/commit/dada32741146148f5e938b7f30dd468f66773eed

**Comment by adrianmcli** [comment]
@CruzMolina that's much better

File: packages/truffle-blockchain-utils/index.js [hunk 2]
```diff
@@ -1,26 +1,14 @@
 const Blockchain = {
   getBlockByNumber(blockNumber, provider, callback) {
-    const params = [blockNumber, true];
     provider.send(
-      {
-        jsonrpc: "2.0",
```
Comment (gnidan) [inline packages/truffle-blockchain-utils/index.js]: We don't need `jsonrpc` / `id`? They get added automatically?

Comment (CruzMolina) [reply packages/truffle-blockchain-utils/index.js]: Ah, looks like we don't actually test any of this code against `geth`.

Comment (CruzMolina) [reply packages/truffle-blockchain-utils/index.js]: Hmm, also need to consider whether an `httpProvider`, `ws` or `ipc` provider is being used...

**Comment by CruzMolina** [comment]
Closing this for now. I *think* the current code suffices for supported `web3` provider types. Turns out this wasn't such an easy win after all!

**Pull request closed without merging**